### DISCUSSION TITLE
History Variables

### DISCUSSION THREAD

**Discussion by MojtabaFarrokh** [ansys/pydyna#452] [Q&A]
Hi
I am asking how to get the history variables in a d3plot file. For instance, there are 10 Elemental History Variables in the following example, but I did not find a clue how I could get them.
 
https://dpf.docs.pyansys.com/version/stable/examples/01-transient_analyses/02-lsdyna_operators.html#lsdyna-operators

**Comment by RobPasMue** [comment]
Hi @MojtabaFarrokh thanks for opening the discussion. @ansys/pydyna team, could you please provide support?

**Comment by RobPasMue** [comment]
Maybe also worth it to ping the @ansys/pydpf-admins since they might be able to answer.

**Comment by zhangzhanqun** [comment]
@MojtabaFarrokh Hi, a sample script have been created to get the history variables in d3plot files, please have a try. any suggestions we can have a discussion.
[d3plot_get_history_var.zip](https://github.com/user-attachments/files/15530457/d3plot_get_history_var.zip)

**Reply by MojtabaFarrokh** [reply]
Thank you, @zhangzhanqun. That worked also for my case.
Is there any documentation about it for more information?

**Comment by zhangzhanqun** [comment]
Hi @MojtabaFarrokh currently, there is no more documentation about this, I will try to create some document about it. if you have any suggestions please tell me.
